src/app/ad-authoring/ad-authoring.component.spec.ts

```diff
@@ -0,0 +1,153 @@
+    const expansion = await loader.getHarness(MatExpansionPanelHarness);
+    await expansion.expand();
+    service = TestBed.inject(AdAuthoringService);
+    spyOn(service, 'updateLandingUrl');
+    spyOn(service, 'updateCallToAction');
+    spyOn(service, 'updateLandingType');
+  });
+
+  it('should create ad-authoring component', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should have correct number of mat selects', async () => {
+    const selects = await loader.getAllHarnesses(MatSelectHarness);
+
+    expect(selects.length).toEqual(2);
+  });
+
+  it('should have the correct number of options for landing type select', async () => {
+    const landingTypeSelect = await loader.getHarness(
+      MatSelectHarness.with({
+        selector: '.landing-page',
+      })
+    );
+
+    await landingTypeSelect.open();
+
+    const typeOptions = await landingTypeSelect.getOptions();
+    expect(typeOptions.length).toBe(3);
+  });
+
+  it('should click the AMP option', async () => {
+    const landingTypeSelect = await loader.getHarness(
+      MatSelectHarness.with({
+        selector: '.landing-page',
+      })
+    );
+    await landingTypeSelect.open();
+
+    await landingTypeSelect.clickOptions({text: 'AMP'});
+
+    const selectedText = await landingTypeSelect.getValueText();
+    expect(selectedText).toBe('AMP');
+  });
+
+  it('should have the correct number of options for call to action select', async () => {
+    const callToActionSelect = await loader.getHarness(
+      MatSelectHarness.with({
+        selector: '.call-to-action',
+      })
+    );
+
+    await callToActionSelect.open();
+
+    const typeOptions = await callToActionSelect.getOptions();
+    expect(typeOptions.length).toBe(21);
+  });
+
+  it('should click the Read Now option', async () => {
+    const callToActionSelect = await loader.getHarness(
+      MatSelectHarness.with({
+        selector: '.call-to-action',
+      })
+    );
+
+    await callToActionSelect.open();
+
+    await callToActionSelect.clickOptions({text: 'Read Now'});
+    const selectedText = await callToActionSelect.getValueText();
+    expect(selectedText).toBe('Read Now');
+  });
+
+  it('should get correct placeholder for landing page url input', async () => {
+    const landingUrlInput = await loader.getHarness(MatInputHarness);
+
+    const placeholder = await landingUrlInput.getPlaceholder();
+
+    expect(placeholder).toBe('Landing Page URL');
+  });
+
+  it('should set value correctly for landing page url input', async () => {
+    const landingUrlInput = await loader.getHarness(MatInputHarness);
+
+    await landingUrlInput.setValue('google.com');
+
+    const value = await landingUrlInput.getValue();
+    expect(value).toBe('google.com');
+  });
+
+  it('changing landing url should call updateLandingUrl function', async () => {
+    const landingUrlInput = await loader.getHarness(MatInputHarness);
+
+    await landingUrlInput.setValue('google.com');
```

Comment:
Is 'google.com' the default value? If so, I'd recommend adding https://google.com as AMP requires it.